src/generators.rs

```diff
@@ -68,6 +68,16 @@ impl GeneratorsChain {
             reader: shake.xof_result(),
         }
     }
+
+    /// Advances the reader n times, squeezing and discarding
+    /// the result.
+    fn fast_forward(&mut self, n: usize) -> &mut Self {
```

Comment:
`skip` and other iterator methods consume self and return self, not the mutable borrows. Let's follow the same style here?

```suggestion
    fn fast_forward(mut self, n: usize) -> Self {
```